ki.js

```diff
@@ -3,21 +3,28 @@
  * Copyright (c) 2015 Denis Ciccale (@tdecs)
  * Released under MIT license
  */
-!function (b, c, d) {
+!(function (b, c, d) {
   /*
    * $ main function
    * a = css selector, dom object, or function
    * http://www.dustindiaz.com/smallest-domready-ever
    * returns instance or executes function on ready
    */
   $ = function (a) {
-    return this instanceof $
-      ? c.push.apply(this, a && a.nodeType ? [a] : '' + a === a ? b.querySelectorAll(a) : d) // constructor
-      : /^f/.test(typeof a) ? /c/.test(b.readyState) ? a() : $(b).on('DOMContentLoaded', a) : new $(a) // function
+    return this instanceof $ // true -> constructor, false -> function
+      ? c.push.apply(
+          this,
+          a && a.nodeType ? [a] : "" + a === a ? b.querySelectorAll(a) : d
+        )
+      : /^f/.test(typeof a)
```

Comment:
Maybe

```js
(typeof a)[0] === "f"
```

could be smaller than 
```js
/^f/.test(typeof a)
```
after compression?

Reply:
Good point, I'll try. Also I think `==` might be enough here, so 1 byte saved non-compressed

Reply:
We use === often enough that it will compress to a single item.

Reply:
We use === one time. So that makes it 2 times then.
I will check with the different options.

Reply:
Your suggestion adds 4 bytes gzipped, from 282 to 286

Reply:
== adds 4 bytes as well, but reduces uncompressed version by 1 bytes.
/^f/ is the second best with 282 gz
/nc/ is the best with 281 gz

Reply:
Ow now I am trying something more: replacing the second `test` with an equality.
With both as `===`, I get 444 / 282
With both as `==`, I get 442 / 281

Reply:
The one thing I am not sure about is if readyState can be undefined. If it can, then `test` is needed

Reply:
Since this is off-topic and unrelated to the current PR, I will close this. I might create another PR, feel free to open an issue to help me track this optimization!